### ISSUE TITLE
Problem (bug?) with path canonicalization on Windows

### ISSUE THREAD

**Issue by ibookstein** [Michael-F-Bryan/mdbook-epub#11]
I tried building TRPL on Windows after adding `[output.epub]` as a renderer to its `book.toml`.
It crashes with the following error:
```
Z:\Projects\rust-lang\book\2018-edition>mdbook build
2018-08-11 12:14:04 [INFO] (mdbook::book): Book building has started
2018-08-11 12:14:04 [INFO] (mdbook::book): Running the epub backend
2018-08-11 12:14:04 [INFO] (mdbook::renderer): Invoking the "epub" renderer
Error: Inspecting the book for additional assets failed
        Caused By: Unable to fetch the canonical path for \\?\Z:\Projects\rust-lang\book\2018-edition\src\img/trpl20-01.png
        Caused By: The filename, directory name, or volume label syntax is incorrect. (os error 123)2018-08-11 12:14:05 [ERROR] (mdbook::renderer): Renderer exited with non-zero return code.
2018-08-11 12:14:05 [ERROR] (mdbook::utils): Error: Rendering failed
2018-08-11 12:14:05 [ERROR] (mdbook::utils):    Caused By: The "epub" renderer failed
```
You'll note the following:

1. The resource file is specified as `![hello from rust](img/trpl20-01.png)` [in the markdown file](https://raw.githubusercontent.com/rust-lang/book/master/2018-edition/src/ch20-00-final-project-a-web-server.md).
2. The path specified in the error message starts with `\\?\`. As stated in [Microsoft's documentation](https://docs.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file):

> File I/O functions in the Windows API convert "/" to "\\" as part of converting the name to an NT-style name, except when using the "\\\\?\\" prefix as detailed in the following sections.

I'm quite new to Rust and thus not sure how where the responsibility falls design-wise (path library or user code), but the regular `mdbook` renderer works fine, so I'm filing this here.

**Comment by Michael-F-Bryan** [comment]
Thanks for the bug report @ibookstein. I'm thinking it would be `mdbook-epub`'s responsibility to convert links into platform-specific paths.

In particular, I think the fault lies somewhere around [here]. This is where we append the link's text (`img/trpl20-01.png`) to the already canonicalized (NT-style) parent directory, so hopefully fixing this is just a case of inserting a `string.replace()` for windows builds.

[here]: https://github.com/Michael-F-Bryan/mdbook-epub/blob/1bf4e32b07bf98e54687006d19ab2de6d6c12542/src/resources.rs#L84

**Comment by ibookstein** [comment]
Shouldn't there be some way to create some sort of relative path object around the string (which would presumably canonize it) and only then join them both, rather than resorting to conditional compilation?

**Comment by nwoltman** [comment]
I just ran into this on Windows, so it seems like `dcf5091` didn't fix the problem.

**Comment by IwateKyle** [comment]
I'm experiencing the same problem. 
.... Unable to fetch the canonical path for \\?\C:\Users....

**Comment by damccull** [comment]
Same here.
```
2020-03-17 23:27:29 [INFO] (mdbook::book): Book building has started
2020-03-17 23:27:29 [INFO] (mdbook::book): Running the epub backend
2020-03-17 23:27:29 [INFO] (mdbook::renderer): Invoking the "epub" renderer
Error: Inspecting the book for additional assets failed
        Caused By: Unable to fetch the canonical path for \\?\C:\Users\David\Projects\rust-lang-book\src\img/trpl20-01.png
        Caused By: The filename, directory name, or volume label syntax is incorrect. (os error 123)
2020-03-17 23:27:29 [ERROR] (mdbook::renderer): Renderer exited with non-zero return code.
2020-03-17 23:27:29 [ERROR] (mdbook::utils): Error: Rendering failed
2020-03-17 23:27:29 [ERROR] (mdbook::utils):    Caused By: The "epub" renderer failed
```

**Comment by blandger** [comment]
Fixed, see comments in [PR#73](https://github.com/Michael-F-Bryan/mdbook-epub/pull/73)
